torch/fx/passes/utils/matcher_utils.py

```diff
@@ -304,6 +306,19 @@ def backtracking(anchor_index, match):
                 match = copy.copy(saved_match)
 
         match = InternalMatch(anchors=self.pattern_anchors)
+
+        # For the unused place holders, if the number of them are the same between pattern and graph,
+        # it's considered as a match and we will add them to the match.nodes_map.
+        graph_placeholder_nodes = [n for n in graph.nodes if n.op == "placeholder"]
+        unused_graph_placeholder_nodes = [placeholder_node for placeholder_node in graph_placeholder_nodes if len(placeholder_node.users) == 0]
+
+        if len(self.unused_pattern_placeholder_nodes) == len(unused_graph_placeholder_nodes):
```

Comment:
I am afraid this is not a general rule that we can accept. 
A pattern graph with unused placeholder is not a well-defined pattern. How should the unused placeholder be interpreted in this case? As it has no consumer, it's effectively a dangling argument can be match with any tensor in the target graph. 
For this very reason, we disallow pattern graph with unused placeholder. If we are missing an assert statement to check this, I should add it.